src/OpenTelemetry/Metrics/HistogramBuckets.cs

```diff
@@ -104,5 +186,18 @@ public bool MoveNext()
                 return false;
             }
         }
+
+        private sealed class Bucket
```

Comment:
Let's name it something else like `BucketLookupNode`. We already have a class name `HistogramBucket` that we use in `HistogramBuckets`. Too many class names ending with `Bucket`. 😂